### ISSUE TITLE
pyAF_introduction stalling during training?

### ISSUE THREAD

**Issue by peterdcotton** [antoinecarme/pyaf#39]
The introductory notebook hangs calling lEngine.train(ozone_dataframe, )

![jypyter_hang](https://cloud.githubusercontent.com/assets/24592491/23774162/69e2bcbe-04f0-11e7-8ce5-b65baa5d0fa2.png)

I am investigating...  

ipykernel==4.5.2
ipython==5.3.0
ipywidgets==6.0.0
jupyter==1.0.0
jupyter-client==5.0.0
jupyter-console==5.1.0
jupyter-core==4.3.0
Keras==1.2.2
matplotlib==2.0.0

**Comment by antoinecarme** [comment]
This example (ozone with 12 forecasts) is a baseline model used in the continuous integration process that you can see here : 

https://travis-ci.org/antoinecarme/pyaf

look for "python tests/func/test_ozone.py" in the logs:

It worked (green) five hours ago (after my two commits). Can you please continue your investigation ? There is something fishy somewhere ...

**Comment by antoinecarme** [comment]
To be sure that pyaf is working properly on your machine (with all software versions installed), you can run locally the build tests : 

 make -f tests/Makefile build-test

This way, you discard all config-related issues.

**Comment by peterdcotton** [comment]
For me pyaf/tests/func/test_ozone.py also hangs after displaying 4 rows

 python3 pyaf/tests/func/test_ozone.py
     Month  Ozone       Time
0  1955-01    2.7 1955-01-01
1  1955-02    2.0 1955-02-01
2  1955-03    3.6 1955-03-01
3  1955-04    5.0 1955-04-01
4  1955-05    6.5 1955-05-01

**Comment by peterdcotton** [comment]
Typical test results are 

`python3  tests/time_res/test_ozone_Daily.py  >  logs/time_res_test_ozone_Daily.log  2>&1
make: [time_res/test_ozone_Daily.py] Error 1 (ignored)
python3 scripts/num_diff.py  tests/references/time_res_test_ozone_Daily.log logs/time_res_test_ozone_Daily.log  >  logs/time_res_test_ozone_Daily.log.diff
tail -10  logs/time_res_test_ozone_Daily.log.diff 
NUM_DIFF_FILES_WITH_DIFFERENT_NUMBER_OF_LINES 196 4
NUM_DIFF_FILES_WITH_TOO_MUCH_DIFF 196 4
NUM_DIFF_NUMBER_OF_DIFFERENT_LINES_AT_LEAST 192
NUM_DIFF_FILES_ARE_DIFFERENT`

**Comment by antoinecarme** [comment]
Peter,

Scikit-learn does not yet fully support python 3.6. Please use python 3.5 until all the dependencies are OK

https://github.com/scikit-learn/scikit-learn/issues/8460

It may work, but it is not sure. Too experimental.

**Comment by antoinecarme** [comment]
This issue will be solved when all required dependencies support python 3.6.

scikit-learn relevant issue is to be followed : 

scikit-learn/scikit-learn#8460

**Comment by antoinecarme** [comment]
Python 3.6 support added (#78). This issue should be closed now. Reopen if needed.

**Comment by anuliketat** [comment]
Hi @antoinecarme 
1. Like I mentioned earlier I have the same problem as the one on this thread. I don't know if the kernel hangs or it is taking too much time to train the data. My data is just a pandas dataframe of two columns with a date time column of 365 days and respective sales column for those dates. I have arranged everything in the exact format as you mentioned in the PyAF documentation and sample code. 
I have attached the snapshot..

![image](https://user-images.githubusercontent.com/11734945/51570514-2fbfcb80-1ec5-11e9-8391-ecd99633ea9c.png)

2. After installing PyAF, I got this error while trying to load some packages - *'tuple object has no attribute type'*. 
I got it resolved after finding that numpy, pandas and sklearn packages got installed again with different versions during the installation of pyAF and other mentioned packages. Also other than those three many other packages didn't load throwing the same error. I don't know which one is doing this.

3. Is there a clear description of what auto forecast engine is doing in the background or is it a black box? How the training is done and what's the metric it takes for validation. Also it would be great if you could provide an article on how auto forecast is doing differently or if better than other popular forecasting techniques like ARIMA, prophet etc.
Thank you.
